### PR TITLE
refactor(ava/advisor): split data-to-advices function and spec generation into sub-files

### PR DESCRIPTION
**Pull request by chenluli**
ava pipeline 改造，分几个阶段进行，该 PR 主要进行文件拆分，将 adviseForChart 函数拆分为 data-processor, spec-processor, spec-generator 几个文件夹，为后续拆分和改造子函数做准备

Refactored the ava/advisor module with significant changes including:
- Dividing the advisor's main function into multiple smaller, more clearly purposed sub-files.
- Separating the spec generation process into its own sub-files to enhance module readability and maintainability.
- The restructured setup aims to simplify future feature expansions and code maintenance.

测试用例已验证
Screenshot 2024-03-26 at 8 48 27 PM

改造后目录结构
Screenshot 2024-03-26 at 9 14 19 PM

### CODE REVIEW

File: packages/ava/package.json [hunk 1]
```diff
@@ -57,6 +57,7 @@
     "@antv/antv-spec": "^0.1.0-alpha.18",
     "@antv/g2": "^5.0.8",
     "@antv/smart-color": "^0.2.1",
+    "@antv/color-schema": "^0.2.3",
```
Comment (BBSQQ) [inline packages/ava/package.json:60]: 哦，用了 color-schema 的 type

File: packages/ava/src/advisor/advise-pipeline/constants.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export const DISCRETE_PALETTE_TYPES = ['monochromatic', 'analogous'] as const;
+export const CATEGORICAL_PALETTE_TYPES = ['polychromatic', 'split-complementary', 'triadic', 'tetradic'] as const;
+export const DEFAULT_COLOR = '#678ef2';
```
Comment (BBSQQ) [inline packages/ava/src/advisor/advise-pipeline/constants.ts:3]: constants 加点儿注释

File: packages/ava/src/advisor/advise-pipeline/score-calculator/score-rules.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  RuleModule,
+  AdvisorOptions,
+  ScoringResultForChartType,
+  ScoringResultForRule,
+  ChartRuleModule,
+  Info,
+} from '../../types';
+
+const defaultWeights = DEFAULT_RULE_WEIGHTS;
+declare type ChartID = (typeof CHART_IDS)[number];
```
Comment (BBSQQ) [inline packages/ava/src/advisor/advise-pipeline/score-calculator/score-rules.ts:14]: ChartID 这个为啥定义在这里？另外我们规范好像应该是 `ChartId`

File: packages/ava/src/advisor/advise-pipeline/index.ts [hunk 4]
```diff
@@ -1 +1,3 @@
-export { advicesForChart } from './advicesForChart';
+export * from './data-to-advices';
+export * from './spec-generator';
+export * from './data-processors';
```
[the remark this answers is not in the document]
Comment (chenluli) [reply packages/ava/src/advisor/advise-pipeline/index.ts:3]: 之前用啥的都有，没规范好 我觉得应该是用横线连接的规范，先把 advisor 内的都改成横线了

File: packages/ava/src/advisor/advise-pipeline/constants.ts (hunk 2 above)
Comment (chenluli) [reply packages/ava/src/advisor/advise-pipeline/constants.ts:6]: 已修改

File: packages/ava/src/advisor/advise-pipeline/score-calculator/score-rules.ts (hunk 3 above)
Comment (chenluli) [reply packages/ava/src/advisor/advise-pipeline/score-calculator/score-rules.ts]: > ChartID 这个为啥定义在这里？另外我们规范好像应该是 `ChartId`

之前遗留问题 已修改

File: packages/ava/src/advisor/advise-pipeline/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (BBSQQ) [reply packages/ava/src/advisor/advise-pipeline/index.ts:3]: 嗯嗯，我感觉横线链接有利于阅读些🤔

**Review by BBSQQ — APPROVED** [review]
LG 拆分 pipeline。重点 CR 了代码规范相关。

**Pull request merged**